reaktive/src/commonTest/kotlin/com/badoo/reaktive/disposable/scope/DisposableScopeImplTest.kt

```diff
@@ -0,0 +1,120 @@
+package com.badoo.reaktive.disposable.scope
+
+import com.badoo.reaktive.base.CompleteCallback
+import com.badoo.reaktive.disposable.Disposable
+import com.badoo.reaktive.test.base.hasSubscribers
+import com.badoo.reaktive.test.completable.TestCompletable
+import com.badoo.reaktive.test.maybe.TestMaybe
+import com.badoo.reaktive.test.observable.TestObservable
+import com.badoo.reaktive.test.single.TestSingle
+import kotlin.test.Test
+import kotlin.test.assertFalse
+import kotlin.test.assertTrue
+
+class DisposableScopeImplTest {
+
+    private val scope = DisposableScopeImpl()
+
+    @Test
+    fun disposes_scoped_disposable_WHEN_disposed() {
+        val disposable = Disposable()
+
+        scope.run { disposable.scope() }
+        scope.dispose()
+
+        assertTrue(disposable.isDisposed)
+    }
+
+    @Test
+    fun completes_scoped_CompleteCallback_WHEN_disposed() {
+        val completeCallback =
```

Comment:
Looks terrible. What is use case?

If you want to have some `onDispose` callback, then maybe it is better to do something like this?
```kotlin
scope.onDispose {
    ...
}
```

Reply:
This is something I'm not sure about. The use case is demonstrated in the sample, where `BehaviorSubject` is created and scoped using `unaryPlus` operator. So when the scope is cancelled - the subject is completed.

Reply:
Then maybe use `Subject` as base?

Reply:
You don't like the ability to complete a completable (something implementing CompletableCallback) whenever scope closes?